src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -2211,7 +2224,9 @@ void GenericThreadStackManagerImpl_OpenThread<ImplClass>::OnDnsBrowseResult(otEr
             // Invoke callback for every service one by one instead of for the whole list due to large memory size needed to
             // allocate on
             // stack.
-            strncpy(dnsResult->mMdnsService.mName, serviceName, sizeof(serviceName));
+            static_assert(ArraySize(dnsResult->mMdnsService.mName) == ArraySize(serviceName),
```

Comment:
While we do expect these to be the same size, in order for this not to break, mName must be the size of serviceName or larger.

Why not test for that then?  With the test as is, I had to go snorkeling around the code to figure out why they need to be the same size, just to find out they actually don't need to be.

```suggestion
            static_assert(ArraySize(dnsResult->mMdnsService.mName) >= ArraySize(serviceName),
```

Reply:
That's fair.  Updated to:
```
            static_assert(ArraySize(dnsResult->mMdnsService.mName) >= ArraySize(serviceName),
                          "The target buffer must be big enough");
```